tfhe/src/integer/bigint/algorithms.rs

```diff
@@ -0,0 +1,240 @@
+use crate::core_crypto::prelude::{Numeric, UnsignedInteger};
+use std::ops::{BitAnd, BitOrAssign, Not, Shl, ShlAssign, Shr, SubAssign};
+
+const BYTES_PER_U64: usize = std::mem::size_of::<u64>() / std::mem::size_of::<u8>();
+
+pub(crate) fn copy_from_le_byte_slice(lhs: &mut [u64], bytes: &[u8]) {
+    assert_eq!(bytes.len(), BYTES_PER_U64 * 4);
```

Comment:
in this form this is u256 specific, I would either prefer a form of const generic or keep this in the u256 file if it's not shared with u512

Same comment for all copy from slice, for the other algorithms you know better than me if they are shared or not between u256 and u512 :)

Reply:
hum, the assert is some left over from moving the code, this thing will work for u512 we just have to change the assert to `assert_eq(lhs.len(), bytes.len())`

Reply:
indeed

Reply:
fixed + test added for u512